staff/maryam-malek/js-arrays/index-of.js

```diff
@@ -0,0 +1,13 @@
+function indexOf(arr, elem) {
+    if(!arr && !elem) throw Error ('function must contain two arguments')
+    if(!(arr instanceof Array)) throw Error (arr + ' is not an array');
+    if(!arr.length) throw Error ('array can not be empty')
+    if(!elem) throw Error ('item to search is missing');
+    var index = -1;
+    for(i=arr.length; i>0; i--){
+        if(arr[i] === elem){
```

Comment:
aquí creo que tienes que devolver sólo el primer índice que cumpla la condición... si acabas el for y hay varios que lo cumplen los devolvería los dos, no?

Reply:
ah! now I see your backward loop! clever solution if you need to iterate all the elements :)
However,  I think that it is easier and more eficiente to make a normal loop and return the first value that you find!